randovania/server/async_race/room_api.py

```diff
@@ -502,7 +520,110 @@ async def submit_proof(sa: ServerApp, sid: str, room_id: int, submission_notes:
     entry.save()
 
 
+async def get_livesplit_url(sa: ServerApp, sid: str, room_id: int) -> str:
+    room = AsyncRaceRoom.get_by_id(room_id)
+    user = await sa.get_current_user(sid)
+    entry = database.AsyncRaceEntry.entry_for(room, user)
+    if entry is None:
+        raise error.NotAuthorizedForActionError
+
+    await sa.ensure_in_room(sid, _get_async_race_socketio_room(room, user))
+
+    token = base64.urlsafe_b64encode(sa.encrypt_str(f"{user.id}/{sid}")).decode("ascii")
+
+    return str(
+        sa.app.url_path_for("livesplit_integration", room_id=room_id, token=token).make_absolute_url(
+            sa.configuration["server_address"]
+        )
+    )
+
+
+_livesplit_event_mapping = {
+    "Reset": AsyncRaceRoomUserStatus.JOINED,
+    "Started": AsyncRaceRoomUserStatus.STARTED,
+    "Paused": AsyncRaceRoomUserStatus.PAUSED,
+    "Resumed": AsyncRaceRoomUserStatus.STARTED,
+    "Finished": AsyncRaceRoomUserStatus.FINISHED,
+}
+
+
+async def emit_async_room_update(sa: ServerApp, room: AsyncRaceRoom, sid_or_user: str | User) -> None:
+    user = await sa.get_current_user(sid_or_user) if isinstance(sid_or_user, str) else sid_or_user
+
+    await sa.sio.emit(
+        signals.ASYNC_RACE_ROOM_UPDATE,
+        (await room.create_session_entry(sa, user)).as_json,
+        namespace="/",
+        to=_get_async_race_socketio_room(room, user),
+    )
+
+
+async def emit_async_room_message(sa: ServerApp, room: AsyncRaceRoom, sid: str, message: str) -> None:
+    pass
```

Comment:
Dead code?

Reply:
Oh, yeah. I gave up on this feature. Was thinking of having the server tell the client of some errors, like "bad data from livesplit received" but didn't figure out the client side of how to display